openmc/geometry.py

```diff
@@ -174,9 +202,9 @@ def get_all_material_universes(self):
         all_universes = self.get_all_universes()
         material_universes = set()
 
-        for universe_id, universe in all_universes.items():
-            cells = universe._cells
-            for cell_id, cell in cells.items():
+        for universe in all_universes:
```

Comment:
As above, `all_universes` is a list of `Universe` so this should be correct. Perhaps the confusion is that the `Geometry` class methods to retrieve primitives return `list` while the `Cell`, `Universe` and `Lattice` class methods all return dictionaries of objects. The original reason for having discrepancy is that traversing the CSG tree and merging dictionaries is much faster than merging lists of unique primitives, while it seems simplest to return lists of the objects from the `Geometry` since most use cases (at least in my experience) may not require the ID keys. Def open to suggestions though if this is an issue for anyone.